### PR TITLE
User can upload files onto scribe.

### PR DESCRIPTION
**Pull request by kennykimjr**
http://scribeit.herokuapp.com/

### CODE REVIEW

File: client/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import UploadFile from './upload.js'
```
Comment (victormejia) [inline client/index.js]: no need for the `.js` part

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  <head>
+    <meta charset="utf-8"/>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.11/semantic.min.css"/>
+    <link rel="stylesheet" href="styles.css"/>
+    <title>Scribe!</title>
+  </head>
+  <body>
+    <div id="app"></div>
+    <script src="https://code.jquery.com/jquery-1.10.2.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.2.11/semantic.min.js"></script>
```
Comment (victormejia) [inline public/index.html]: you won't need semantic or jquery here

File: server.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    unique_id: req.file.filename,
+    parent_id: 'none'
+  })
+  .then(response => {
+    return res.sendStatus(201)
+  })
+  .catch(error => {
+    console.log(error)
+    return res.sendStatus(500)
+  })
```
Comment (victormejia) [inline server.js]: extract this database transaction into a function

File: server.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+function addRecord(fileName, hash) {
+  return knex('history').insert({
+    file_name: fileName,
+    unique_id: hash,
+    parent_id: 'none'
+  })
+}
+
+app.post('/file', upload.single('file'), (req, res) => {
+  addRecord(req.file.originalname, req.file.filename)
```
Comment (victormejia) [inline server.js]: hmmm, shouldn't you generate a unique hash here?

File: migrations/1502135816428_table.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
+-- up
+CREATE TABLE IF NOT EXISTS history (
+  file_name text,
+  parent_id text,
+  unique_id text,
```
Comment (victormejia) [inline migrations/1502135816428_table.sql]: you shouldn't need a parent_id. Revisions will be kept in a revisions table

File: server.js (hunk 4 above)
Comment (kennykimjr) [inline server.js]: This automatically generates unique hashes. I've seen this with multiple uploads of the same file, they all have unique hashes automatically.

File: migrations/1502135816428_table.sql (hunk 5 above)
Comment (victormejia) [inline migrations/1502135816428_table.sql]: gotcha!

File: server.js (hunk 4 above)
Comment (kennykimjr) [inline server.js]: screen shot 2017-08-10 at 10 41 12 am

**Pull request merged**